crypto/ledger/ledger_real.go

```diff
@@ -3,18 +3,30 @@
 
 package ledger
 
-import ledger "github.com/cosmos/ledger-cosmos-go"
+import (
+	"github.com/cosmos/cosmos-sdk/crypto/keys/secp256k1"
+	"github.com/cosmos/cosmos-sdk/crypto/types"
+	ledger "github.com/cosmos/ledger-cosmos-go"
+)
 
 // If ledger support (build tag) has been enabled, which implies a CGO dependency,
 // set the discoverLedger function which is responsible for loading the Ledger
 // device at runtime or returning an error.
 func init() {
-	discoverLedger = func() (SECP256K1, error) {
+	options.discoverLedger = func() (SECP256K1, error) {
 		device, err := ledger.FindLedgerCosmosUserApp()
 		if err != nil {
 			return nil, err
 		}
 
 		return device, nil
 	}
+
+	// Set default values for Cosmos Ledger instance. These can be updated
+	// by setting fields in the Keyring Options.
+	options.createPubkey = func(key []byte) types.PubKey {
+		return &secp256k1.PubKey{Key: key}
+	}
+	options.appName = "Cosmos"
+	options.skipDERConversion = false
```

Comment:
There's two code blocks that do the exact same thing -- I would move this setup into a helper function.

Reply:
Revised